src/sentry/api/endpoints/api_token_details.py

```diff
@@ -0,0 +1,71 @@
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import never_cache
+from rest_framework import serializers
+from rest_framework.fields import CharField
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.api.endpoints.api_tokens import _get_appropriate_user_id
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models.apitoken import ApiToken
+
+ALLOWED_FIELDS = ["name", "tokenId"]
+
+
+class ApiTokenNameSerializer(serializers.Serializer):
+    name = CharField(max_length=255, allow_blank=True, required=True)
+
+
+@control_silo_endpoint
+class ApiTokenDetailsEndpoint(Endpoint):
```

Comment:
i would recommend adding a `convert_args` function to this that fetches the token from the `token_id`, then you can pass the resource directly into the GET/PUT methods here. example: https://github.com/getsentry/sentry/blob/4652fc7032c182ba8ce50daa64cf39766b32158e/src/sentry/api/endpoints/project_details.py#L473

also, should we move the DELETE method here as well?

Reply:
added DELETE method (will need to change UI to call this details endpoint later)